tests/acceptance/features/bootstrap/WebDav.php

```diff
@@ -456,28 +456,35 @@ public function makeDavRequest(
 	 * @param string|null $height
 	 *
 	 * @return void
+	 * @throws GuzzleException
 	 */
 	public function downloadPreviews(string $user, ?string $path, ?string $doDavRequestAsUser, ?string $width, ?string $height):void {
-		$user = $this->getActualUsername($user);
-		$doDavRequestAsUser = $this->getActualUsername($doDavRequestAsUser);
 		$urlParameter = [
 			'x' => $width,
 			'y' => $height,
-			'forceIcon' => '0',
 			'preview' => '1'
 		];
-		$this->response = $this->makeDavRequest(
+		$url = $this->getBaseUrl() . '/remote.php';
+		$davVersion = $this->getDavPathVersion();
+		if ($davVersion === WebDavHelper::DAV_VERSION_SPACES) {
+			$spaceId = $this->getPersonalSpaceIdForUser($doDavRequestAsUser ?? $user);
+			$url .= '/dav/spaces/' . $spaceId . '/';
+		} elseif ($davVersion === WebDavHelper::DAV_VERSION_NEW) {
+			$url .= '/dav/files/' . ($doDavRequestAsUser ?? $user) . '/';
+		} else {
+			$url .= '/webdav/';
+		}
+
+		$urlParameter = \http_build_query($urlParameter, '', '&');
+		$path .= '?' . $urlParameter;
+		$fullUrl = WebDavHelper::sanitizeUrl($url . $path);
```

Comment:
Does `makeDavRequest` not do all of this? if we need to add `remote.php` in the pathe then maybe we can add a option in the `makeDavRequest` to optionally add it? or am I missing something

Reply:
I think the full URL is made based on the `dav` we choose. @amrita-shrestha what was wrong with the old function?